entity_event_generator.tf

```diff
@@ -0,0 +1,56 @@
+locals {
+  entity_event_lambda_name = "${local.environment}-entity-event-lambda"
+  entity_event_topic_name  = "${local.environment}-preservica-config-topic"
+  last_polled_table_name   = "${local.environment}-entity-event-last-updated"
```

Comment:
I've gone with `${local.environment}-entity-event-lambda-query-start-datetime` because I used to be a Java programmer and long names are always fun.

Reply:
but we are storing the latest event action date now, right; shouldn't it be names something that indicates this? What do we mean be "start-datetime"?

Reply:
I read `last-updated` as storing the value of the datetime of most recent updated entity in Preservica. This isn't always going to be the case as we could have a massive backlog of entity updates in Preservica where the lambda will need to run multiple times before it's caught up.

I suggested `query-start-datetime` as this is the datetime we enter into our query to Preservica as the updates from datetime. Maybe `query-from-datetime` would be more suitable?

Reply:
Ohh, I understand now; then `start-datetime` makes sense. How about `event-actions-query-start-datetime` or `event-actions-request-start-datetime`? It's longer but it's more specific which is better to be since this is a table name.

Reply:
`${local.environment}-entity-event-lambda-event-actions-query-start-datetime` seems ambiguous as we don't query the event actions to generate events. If anything it would be `${local.environment}-entity-event-lambda-updated-since-query-start-datetime`. But I wonder if that's too detailed as we're now bringing Preservica specific naming into the IaC, is the `updated-since` endpoint implied as we know that lambda is generating events via a query to the system?

Reply:
_We_ know, but for anyone else (including our future selves), it's ambiguous. To me, `updated-since` isn't really Preservica-specific because to me, it describes the query that we are making; if we switched to another provider but had to make the same request, it would still make sense.

Reply:
It's a good job copy and paste exists.

Reply:
I guess we should also add `generator` to the long name as well for good measure.

Reply:
I would worry for anyone who can't work out what the table is for based on what it currently is. I think we're fine.

Reply:
`${local.environment}-entity-event-lambda-table-to-hold-updated-since-query-start-datetime-query-for-the-purpose-of-retrieving-entities-updated-since-specified-datetime`

Reply:
`${local.environment}-entity-event-lambda-table-to-hold-updated-since-query-start-datetime-query-for-the-purpose-of-retrieving-entities-updated-since-specified-datetime`

Reply:
I'd say`${local.environment}-entity-event-lambda-table-to-hold-updated-since-datetime-value-for-use-as-query-for-the-purpose-of-retrieving-entities-updated-since-specified-datetime`